### DISCUSSION TITLE
Wayland and xdotool

### DISCUSSION THREAD

**Discussion by sughero** [bulletmark/libinput-gestures#368] [Q&A]
It's well known that xdotool was meant for x11-xorg and not for Wayland, and has some issues with Wayland.  [They suggest using dotool or ydotool instead](https://github.com/jordansissel/xdotool).
On the other side all configurations files for libinput-gestures are using xdotool in order to recall a keyboard shortcut instead of launching directly a command as widely showed [here.](https://github.com/bulletmark/libinput-gestures)
I am aware that in [other discussions](https://github.com/bulletmark/libinput-gestures/discussions/365) ydotools has been proposed as an alternative.
Since Wayland is expected to 'soon' replace Xorg, I wonder if dropping xdotool in favor of something else is part of the plan.

**Comment by bulletmark** [comment]
Your first link points to the ydotool README but should point to the Wayland section there. Your second link points to this libinput-gestures repo so just seems wrong. My opinion stated in that 3rd link has not changed. Also, you wrote all this but didn't mention anything about the [section on Wayland here in the README](https://github.com/bulletmark/libinput-gestures#wayland-and-other-notes) which discusses this issue.

You say Wayland is expected "soon" to replace Xorg but note that Wayland became the default for GNOME at 3.22, i.e. 8 years ago! I have been using that default GNOME  since then and have continued to use xdotool just fine.

**Comment by sughero** [comment]
Okay I frankly apologize. I completely messed up with all the links in my previous post. 
Moreover, if I had dug  more deeply and read all the way through Wayland's comment in the README, I would have avoided saying what I said.
I was misled by this statement inside the README

![image](https://github.com/user-attachments/assets/b0c86ddd-6408-41ad-91fe-05b1b0f0df37)
where I admit I took Wayland for **X**Wayland.

When I installed libtool-gestures, I expected something that would work out of the box. 
Instead, it took me a while to realize that if pinch-to-zoom wasn’t working, and spreading two fingers on the touchpad (for zooming) opened strange windows, the culprit wasn’t libtool-gestures or how the gesture was recognized, but rather the fact that xdotool doesn’t work with Wayland.

**Reply by bulletmark** [reply]
I wrote this tool as a quick "hack" about 10 years ago (pre-Wayland) and as the README still says it was "only intended to be used temporarily until GNOME and other DE's action gestures natively". Since then, GNOME at least has added some native gestures but still no where as flexible or as expansive as what you can accomplish using libinput-gestures so I (and many others obviously) have continued to use it. I have also added quite a few extra features over the years. Unfortunately there have been a few obstacles placed in the way over time (Wayland and it's strict security model being the biggest one!) so it does require the user to understand the issues and accommodate their desired configuration so yes, it is unfortunately not "plug and play".
